addon/utils/locale.js

```diff
@@ -87,10 +87,11 @@ export default class Locale {
   }
 
   _defineMissingTranslationTemplate(key) {
+    const I18n = this.container.lookupFactory('service:i18n');
     const missingMessage = this.container.lookupFactory('util:i18n/missing-message');
     const locale = this.id;
 
-    function missingTranslation(data) { return missingMessage(locale, key, data); }
+    function missingTranslation(data) { return missingMessage.call(I18n, locale, key, data); }
```

Comment:
I'm probably not nearly familiar enough with the i18n API... but I don't see the point in allowing to call buildMissingTranslation on i18n object.
BTW, won't this cause a recursion loop if called from the missing-message function, i.e:
```export default function(locale, key, data) {
  return this.buildMissingTranslation(locale, key, data)
}```